src/composables/useNavigationGuards.ts

```diff
@@ -13,6 +17,53 @@ export default function useNavigationGuards() {
   const { setShowRedirectModal, isVeBalSupported } = useVeBal();
   const { setSidebarOpen } = useSidebar();
 
+  router.beforeEach((to, from, next) => {
+    const betaEnv =
+      window.location.host.split('.')[0] === 'beta' ||
+      window.location.host.split('.')[0] === 'staging';
+    const subdomain = betaEnv
+      ? window.location.host.split('.')[1]
+      : window.location.host.split('.')[0];
+    const networkFromSubdomain = networkFromSlug(subdomain);
+    if (networkFromSubdomain) {
+      // old format subdomain redirect
+      const networkFromUrl = networkFromSlug(to.params.networkSlug?.toString());
+      localStorage.setItem(
+        'networkId',
+        (networkFromUrl || networkFromSubdomain).toString()
+      );
+      window.location.href = networkFromUrl
+        ? `${appUrl()}${to.fullPath}`
+        : `${appUrl()}/${config[networkFromUrl || networkFromSubdomain].slug}${
+            to.fullPath
+          }`;
+    } else {
+      // check for network in url and redirect if necessary
+      const networkSlug = to.params.networkSlug?.toString();
+      if (networkSlug) {
+        const networkFromUrl = networkFromSlug(networkSlug);
+        const localStorageNetwork: Network = networkFor(
+          localStorage.getItem('networkId') ?? '1'
+        );
+        if (!networkFromUrl) {
+          // missing or incorrect network name -> next() withtout network change
+          next();
+        } else if (localStorageNetwork === networkFromUrl) {
+          // if on the correct network -> next()
+          next();
+        } else {
+          // if on different network -> update localstorage and reload
+          document.write('');
+          localStorage.setItem('networkId', networkFromUrl.toString());
+          window.location.href = `/#${to.fullPath}`;
+          router.go(0);
+        }
```

Comment:
Again, I think this should be extracted into a function and write tests to be sure it does as intended based on some test URLs.

Reply:
See above.